### PR TITLE
feat: multiple clusters and independent container registry

### LINKED ISSUE #286: Test kind multi-cluster
Test the kind deployment script for multi-cluster deployments.
- [x] multiple ingress controllers and port mappings

### LINKED ISSUE #287: kind: Move the image registry creation to an independent command

### LINKED ISSUE #308: Error calling kind script
```
bash <(curl -s -L https://raw.githubusercontent.com/snowdrop/k8s-infra/main/kind/kind.sh)                                                                                                  INT ✘  18:01:36

/dev/fd/13: line 877: [: -v: unary operator expected
/dev/fd/13: line 881: [: -v: unary operator expected
/dev/fd/13: line 885: [: -v: unary operator expected
/dev/fd/13: line 889: [: ==: unary operator expected

Welcome to our

   _____                                  _
  / ____|                                | |
 | (___    _ __     ___   __      __   __| |  _ __    ___    _ __
  \___ \  | '_ \   / _ \  \ \ /\ / /  / _  | |  __|  / _ \  | \ _ \
  ____) | | | | | | (_) |  \ V  V /  | (_| | | |    | (_) | | |_) |
 |_____/  |_| |_|  \___/    \_/\_/    \__,_| |_|     \___/  |  __/
                                                            | |
                                                            |_|
Script to create/delete a Kubernetes cluster using kind like a :
- (secured) docker container registry
- ingress controller (nginx, kourier)

/dev/fd/13: line 225: [: ==: unary operator expected
✔ Pre requisites check passed!
```

### PR DESCRIPTION
**Pull request by jacobdotcosta**
Multiple improvements:
* kind multi-cluster
* kind: Move the image registry creation to an independent command

Closes #286 
Closes #287 
Closes #308

### CODE REVIEW

File: kind/kind.sh [hunk 1]
```diff
@@ -862,9 +898,11 @@ while [ $# -gt 0 ]; do
         shift
     else
         case $1 in
-            install) COMMAND="install" ;;
-            remove) COMMAND="remove" ;;
-            *) INVALID_SWITCH="${INVALID_COMMAND} $1" ; break 2 ;;
+            install-cluster) COMMAND="install-cluster" ;;
+            remove-cluster) COMMAND="remove-cluster" ;;
+            install-registry) COMMAND="install-registry" ;;
+            remove-registry) COMMAND="remove-registry" ;;
```
Comment (cmoulliard) [inline kind/kind.sh:904]: Could we use instead `./kind.sh install|remove cluster|registry ?

To be discussed: If the script is finally used to manage 2 different things, should we then continue to manage everything part of the kind script or to have 2 scripts: kind.sh and registry.sh.

Comment (cmoulliard) [reply kind/kind.sh:904]: Another issue is that if you do that, then you will break the existing usage that I already integrated in some of my github repo to use/install kind top of TAP, Red Hat Buildpaks, DAPR, etc

Comment (jacobdotcosta) [reply kind/kind.sh:904]: > registry

The only way this wouldn't happen is having an independent script for the container registry.

**Comment by cmoulliard** [comment]
Ideally, each ticket should be part of separate PRs as they cover different concepts to be reviewed/tested @jacobdotcosta

**Comment by jacobdotcosta** [comment]
> Ideally, each ticket should be part of separate PRs as they cover different concepts to be reviewed/tested @jacobdotcosta

I agree, but since other problems arose which required fixing, and this PR was taking some time to be ready, I decided to have them included.

File: kind/README.adoc [hunk 2]
```diff
@@ -111,41 +126,97 @@ Choices:
 
 == Install
 
-Creating a kubernetes cluster is as simple as executing the following command.
+This script allows creating both a kind cluster with associated ingress controller and a container registry in separate commands.
```
Comment (cmoulliard) [inline kind/README.adoc:129]: Why do you say ".... in separate commands" ?
If this PR installs now using separate parameters the kind cluster AND/OR registry, why, when I execute this command (using bash PR script) -> "curl -s https://raw.githubusercontent.com/snowdrop/k8s-infra/7b7186a8fd90d38195520dc322d1e708422f1f4e/kind/registry.sh | bash -s" nothing is displayed about the commands to be used to install the container registry ?

Comment (jacobdotcosta) [reply kind/README.adoc:129]: I see what you mean, the help file still refers to the kind cluster. I'll fix that.

```
curl -s https://raw.githubusercontent.com/snowdrop/k8s-infra/7b7186a8fd90d38195520dc322d1e708422f1f4e/kind/registry.sh | bash -s
❌ A command must be provided!!!

Usage: 
	./registry.sh command [parameters,...]

Available commands: 
	install					Install the kind cluster
	remove					Remove the kind cluster

Parameters: 
	-h, --help				This help message

	--provider <provider>			Container Runtime [docker,podman]. Default: docker
	--registry-image-version <version>	Version of the registry container to be used. Default: 2.6.2
	--registry-name <name>			Name of the registry. Default: <cluster_name>-registry
	--registry-password <password>		Registry user password. Default: snowdrop
	--registry-port <port>			Port of the registry. Default: 5000
	--registry-user <user>			Registry user. Default: admin
	--secure-registry			Secure the docker registry. Default: No
	-v, --verbosity <value>			Logging verbosity (0..9). Default: 1
						A verbosity setting of 0 logs only critical events.
```

Comment (jacobdotcosta) [reply kind/README.adoc:129]: Fixed.

```
./kind/registry.sh                                                                                                                       several-fixes 
❌ A command must be provided!!!

Usage: 
	./registry.sh command [parameters,...]

Available commands: 
	install					Install the container registry
	remove					Remove the container registry

Parameters: 
	-h, --help				This help message

	--provider <provider>			Container Runtime [docker,podman]. Default: docker
	--registry-image-version <version>	Version of the registry container to be used. Default: 2.6.2
	--registry-name <name>			Name of the registry. Default: kind-registry
	--registry-password <password>		Registry user password. Default: snowdrop
	--registry-port <port>			Port of the registry. Default: 5000
	--registry-user <user>			Registry user. Default: admin
	--secure-registry			Secure the docker registry. Default: No
	-v, --verbosity <value>			Logging verbosity (0..9). Default: 1
						A verbosity setting of 0 logs only critical events.

```

@cmoulliard

**Pull request merged**